modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/loader/java/JavaErrorParsingTestCase.java

```diff
@@ -124,13 +129,30 @@ public void operationThrowsOverridesExtensionThrows() {
   }
 
   @Test
-  public void operationInheritsExtensionErrorThrows() {
+  public void operationAndConstructInheritExtensionErrorThrows() {
     extensionModel = loadExtension(HeisenbergWithExtensionThrows.class);
 
     OperationModel someOperation = extensionModel.getOperationModel("someOperation").get();
     Optional<ErrorModel> operationError = someOperation.getErrorModels().stream()
         .filter(errorModel -> errorModel.getType().equals(EXTENSION.getType())).findFirst();
     assertThat(operationError.isPresent(), is(true));
+
+    // W-12289050
+    ConstructModel someConstruct = extensionModel.getConstructModel("someConstruct").get();
+    Optional<ErrorModel> constructError = someConstruct.getErrorModels().stream()
+        .filter(errorModel -> errorModel.getType().equals(EXTENSION.getType())).findFirst();
```

Comment:
why does the construct get the EXTENSION type?

Reply:
Shouldn't it behave like the any operation in that sense?